misc/python/materialize/parallel_workload/action.py

```diff
@@ -645,7 +648,10 @@ def run(self, exe: Executor) -> None:
                 return
             source_id = self.db.source_id
             self.db.source_id += 1
-            cluster = self.rng.choice(self.db.storage_clusters)
+            cluster = self.rng.choice(self.db.clusters)
+            if len(cluster.replicas) > 1:
+                # cannot create source in cluster with more than one replica
+                return
```

Comment:
Not deeply familiar with the implementation, but here it seems like we're probabilistically creating a new source, which previously was deterministic. Is this the intended behavior? Alternatively, we could randomly select from a cluster that only has a single replica, and return or panic if there's none.

Reply:
Yes, it's probabilistic now, so we will have fewer sources. Actually let me change the logic so we always have 1 cluster with a single replica around, makes it easier to reason about this.